scripts/utils/migrate/taxonomy-redirects.js

```diff
@@ -29,24 +29,42 @@ const getTaxonomyPath = async (uri) => {
   }
 };
 
+const getNodePath = async (uri) => {
+  const id = uri.replace(/.*\/node\//, '');
+  const url = new URL(`/api/migration/content/node/${id}/urlPath`, BASE_URL);
+  const headers = {
+    'Content-Type': 'application/json',
+    'X-Phpshield-Key-Disable': process.env.ACQUIA_DEV_PHP_SHIELD_KEY,
+  };
+  try {
+    const res = await fetch(url, { headers });
+    const json = await res.json();
+    return get(json, 'docs[0].doc.path');
+  } catch (error) {
+    return null;
+  }
+};
+
 const fetchTaxonomyRedirects = async (redirects) => {
   const taxonomy = await Promise.all(
     Object.entries(redirects)
       .filter(([uri]) => uri.startsWith('/taxonomy'))
       .map(async ([url, paths]) => [await getTaxonomyPath(url), paths])
   );
 
-  return taxonomy.reduce((memo, [url, paths]) => {
-    const filteredPaths = paths.filter(
-      (path) =>
-        path !== url &&
-        !path.startsWith('/node') &&
-        !path.startsWith('/taxonomy')
+  return await taxonomy.reduce(async (memo, [url, paths]) => {
+    const filteredPaths = await Promise.all(
+      paths
+        .filter((path) => path !== url && !path.startsWith('/taxonomy'))
+        .map(async (path) =>
+          path.startsWith('/node') ? await getNodePath(path) : path
+        )
+        .filter((path) => Boolean(path))
```

Comment:
Since you're returning a list of Promises from the `map` right above this, this `filter` is going to return `true` 100% of the time (since `Boolean(Promise<any>) === true`). You'll need to run this `filter` after you've `await`ed the results from the `Promise.all`